consensus/types/src/blob_sidecar.rs

```diff
@@ -171,31 +169,6 @@ impl<T: EthSpec> BlobSidecar<T> {
         // Fixed part
         Self::empty().as_ssz_bytes().len()
     }
-
-    // this is mostly not used except for in testing
-    pub fn sign(
-        self: Arc<Self>,
-        secret_key: &SecretKey,
-        fork: &Fork,
-        genesis_validators_root: Hash256,
-        spec: &ChainSpec,
-    ) -> SignedBlobSidecar<T> {
-        let signing_epoch = self.slot.epoch(T::slots_per_epoch());
-        let domain = spec.get_domain(
-            signing_epoch,
-            Domain::BlobSidecar,
-            fork,
-            genesis_validators_root,
-        );
-        let message = self.signing_root(domain);
-        let signature = secret_key.sign(message);
-
-        SignedBlobSidecar {
-            message: self,
-            signature,
-            _phantom: PhantomData,
-        }
-    }
```

Comment:
Moved to `Sidecar` trait, so we can sign over blinded blobs as well in tests.

https://github.com/sigp/lighthouse/pull/4719/files#diff-55c0e2e9924a444c0db563e5394af77f38b98c6a85e5c865816764003739516eR45-R68